bansa/src/test/kotlin/com/brianegan/bansa/ReselectTest.kt

```diff
@@ -0,0 +1,409 @@
+        assertThat(selector(state1)).isEqualTo(3)
+        assertThat(selector.recomputations).isEqualTo(1)
+        val state2=StateAB(a=3,b=2)
+        assertThat(selector(state2)).isEqualTo(5)
+        assertThat(selector(state2)).isEqualTo(5)
+        assertThat(selector.recomputations).isEqualTo(2)
+    }
+
+    data class StateSubStateA (val sub:StateA)
+    @Test
+    fun MemoizedCompositeArgumentsTest() {
+        val selector = SelectorFor<StateSubStateA>()
+                .withField {sub}
+                .compute {sub:StateA -> sub}
+        val state1= StateSubStateA(StateA(1))
+        assertThat(selector(state1)).isEqualTo(StateA(1))
+        assertThat(selector(state1)).isEqualTo(StateA(1))
+        assertThat(selector.recomputations).isEqualTo(1)
+        val state2= StateSubStateA(StateA(2))
+        assertThat(selector(state2)).isEqualTo(StateA(2))
+        assertThat(selector.recomputations).isEqualTo(2)
+    }
+    @Ignore //not implemented
+    @Test
+    fun FirstArgumentCanBeAnArrayTest() {
+        /*
+            const selector = createSelector(
+      [ state => state.a, state => state.b ],
+      (a, b) => {
+        return a + b
+      }
+    )
+    assert.equal(selector({ a: 1, b: 2 }), 3)
+    assert.equal(selector({ a: 1, b: 2 }), 3)
+    assert.equal(selector.recomputations(), 1)
+    assert.equal(selector({ a: 3, b: 2 }), 5)
+    assert.equal(selector.recomputations(), 2)
+         */
+    }
+
+    //TODO I have not really implemented props, but not sure what is the use case.
+    @Test
+    fun CanAcceptPropsTest() {
+
+        val selector = SelectorFor<StateAB>()
+                .withField {a}
+                .withField {b}
+                .withField {100}
+                .compute{a:Int,b:Int,c:Int -> a+b+c}
+        val state1=StateAB(a=1,b=2)
+        assertThat(selector(state1)).isEqualTo(103)
+
+        /*
+       let called = 0
+       const selector = createSelector(
+         state => state.a,
+         state => state.b,
+         (state, props) => props.c,
+         (a, b, c) => {
+           called++
+           return a + b + c
+         }
+       )
+       assert.equal(selector({ a: 1, b: 2 }, { c: 100 }), 103)
+            */
+    }
+
+    @Test
+    fun ChainedSelectorTest() {
+        val selector1= SelectorFor<StateSubStateA>()
+                .withField {sub}
+                .compute{sub:StateA -> sub}
+        val selector2= SelectorFor<StateSubStateA>()
+                .withSelector(selector1)
+                .compute{sub:StateA -> sub.a }
+        val state1=StateSubStateA(StateA(1))
+        assertThat(selector2(state1)).isEqualTo(1)
+        assertThat(selector2(state1)).isEqualTo(1)
+        assertThat(selector2.recomputations).isEqualTo(1)
+        val state2=StateSubStateA(StateA(2))
+        assertThat(selector2(state2)).isEqualTo(2)
+        assertThat(selector2.recomputations).isEqualTo(2)
+    }
+    @Ignore //not implemented
```

Comment:
I'd recommend removing any of these not implemented tests. If you want do document missing methods in a todo, that'd be cool, but it feels messy to leave a bunch of dead code lying around.